src/navier_stokes/INSVCStaggeredConservativeMassMomentumSSPRKIntegrator.cpp

```diff
@@ -0,0 +1,531 @@
+                        a2 = 0.25;
+                        break;
+                    }
+                    else
+                    {
+                        TBOX_ERROR("This statement should not be reached");
+                        break;
+                    }
+                case 2:
+                    a0 = 1.0 / 3.0;
+                    a1 = 2.0 / 3.0;
+                    a2 = 2.0 / 3.0;
+                    break;
+                default:
+                    TBOX_ERROR("This statement should not be reached");
+                }
+                computeDensityUpdate(R_new_data,
+                                     a0,
+                                     R_cur_data,
+                                     a1,
+                                     R_pre_data,
+                                     a2,
+                                     V_adv_data,
+                                     R_half_data,
+                                     R_src_data,
+                                     side_boxes,
+                                     dt,
+                                     dx);
+
+                Pointer<SideData<NDIM, double> > V_cur_data = patch->getPatchData(d_V_current_idx);
+                if ((d_density_time_stepping_type == SSPRK2 && step == 1) ||
+                    (d_density_time_stepping_type == SSPRK3 && step == 2))
+                {
+                    computeErrorOfMassConservationEquation(
+                        E_data, R_new_data, R_cur_data, V_adv_data, R_half_data, side_boxes, dt, dx);
+
+                    // subtract Error*U from the convective operator.
+                    for (unsigned int axis = 0; axis < NDIM; ++axis)
+                    {
+                        for (Box<NDIM>::Iterator it(SideGeometry<NDIM>::toSideBox(patch_box, axis)); it; it++)
+                        {
+                            SideIndex<NDIM> si(it(), axis, SideIndex<NDIM>::Lower);
+
+                            (*N_data)(si) -= (*V_cur_data)(si) * (*E_data)(si);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Refill boundary values of newest density
+    const double new_time = d_current_time + dt;
+    std::vector<InterpolationTransactionComponent> new_transaction_comps(1);
+    new_transaction_comps[0] = InterpolationTransactionComponent(d_rho_scratch_idx,
+                                                                 d_rho_new_idx,
+                                                                 "CONSERVATIVE_LINEAR_REFINE",
+                                                                 false,
+                                                                 "CONSERVATIVE_COARSEN",
+                                                                 d_density_bdry_extrap_type,
+                                                                 false,
+                                                                 d_rho_bc_coefs);
```

Comment:
FWIW, notice that these can be initialized much more compactly with C++11 --- I think this should work:
```suggestion
    std::vector<InterpolationTransactionComponent> new_transaction_comps({{d_rho_scratch_idx,
                                                                                                                                      d_rho_new_idx,
                                                                                                                                      "CONSERVATIVE_LINEAR_REFINE",
                                                                                                                                      false,
                                                                                                                                      "CONSERVATIVE_COARSEN",
                                                                                                                                      d_density_bdry_extrap_type,
                                                                                                                                      false,
                                                                                                                                      d_rho_bc_coefs}});
```

Reply:
(Much might be a little bit of an exaggeration. 🤷‍♂️ )